disco_aws_automation/disco_autoscale.py

```diff
@@ -54,17 +54,21 @@ def get_launch_config_name(self, hostclass):
 
     def _filter_by_environment(self, items):
         """Filters autoscaling groups and launch configs by environment"""
-        return [
-            item for item in items
-            if item.name.startswith("{0}_".format(self.environment_name))
-        ]
+        for item in items:
+            try:
+                if item.name.startswith("{0}_".format(self.environment_name)):
+                    yield item
```

Comment:
are we sure that the returned items will only be iterated over once?

Reply:
Looking at the usage, looks like yes.